src/renderers/dom/fiber/ReactDOMFiber.js

```diff
@@ -88,8 +88,24 @@ var DOMRenderer = ReactFiberReconciler({
     return document.createTextNode(text);
   },
 
-  commitTextUpdate(textInstance : TextInstance, oldText : string, newText : string) : void {
+  commitTextUpdate(textInstance : TextInstance, oldText : string, newText : string, parentInstance: Instance) : void {
+    let nodeValueLen = textInstance.nodeValue.length;
+
+    // The Node may have been split in this case we need to clean up some next sibling nodes
+    if (nodeValueLen < oldText.length) {
+      let nextSibling = textInstance.nextSibling;
+      while (nextSibling instanceof Text && nodeValueLen < oldText.length) {
```

Comment:
Well, after thinking about it, I doubt it is useful! Let me see if I can simplify all of this!